### PR TITLE
vmlatency, parser, test: Apply test-case pattern

### PR DESCRIPTION
**Pull request by ormergi**
Follow "test-case" pattern similar to other test suites in the code base in order to make it easier to add new tests.

Signed-off-by: Or Mergi <ormergi@redhat.com>

### CODE REVIEW

File: checkups/kubevirt-vm-latency/vmlatency/internal/latency/pingparser_test.go [hunk 1]
```diff
@@ -28,9 +28,7 @@ import (
 	"github.com/kiagnose/kiagnose/checkups/kubevirt-vm-latency/vmlatency/internal/latency"
 )
 
-func TestParsePingResults(t *testing.T) {
-	const (
-		pingOutput = `
+const pingOutput = `
```
Comment (orelmisan) [inline checkups/kubevirt-vm-latency/vmlatency/internal/latency/pingparser_test.go:31]: `pingOutput` could be moved to `TestParsePingResults()` in order to decrease its scope.

File: checkups/kubevirt-vm-latency/vmlatency/internal/latency/pingparser_test.go [hunk 2]
```diff
@@ -42,24 +40,33 @@ PING 1.1.1.1 (1.1.1.1) 56(84) bytes of data.
...
+var pingResults = latency.Results{
+	Min:         1732000 * time.Nanosecond,
+	Average:     2074000 * time.Nanosecond,
+	Max:         2382000 * time.Nanosecond,
+	Time:        4004 * time.Millisecond,
+	Transmitted: 5,
+	Received:    5,
+}
+
+type pingParserTestCase struct {
```
Comment (orelmisan) [inline checkups/kubevirt-vm-latency/vmlatency/internal/latency/pingparser_test.go:53]: `pingParserTestCase` could be moved into `TestParsePingResults()` in order to decrease its scope.

File: checkups/kubevirt-vm-latency/vmlatency/internal/latency/pingparser_test.go [hunk 3]
```diff
@@ -42,24 +40,33 @@ PING 1.1.1.1 (1.1.1.1) 56(84) bytes of data.
...
+			pingOutput:      pingOutput,
+			expectedResults: pingResults,
+		},
 	}
-	if expectedResults.Max, err = time.ParseDuration("2.382ms"); err != nil {
-		panic(err)
+	for _, testCase := range testCases {
+		t.Run(testCase.description, func(t *testing.T) {
+			assert.Equal(t, testCase.expectedResults, latency.ParsePingResults(testCase.pingOutput))
+		})
```
Comment (orelmisan) [inline checkups/kubevirt-vm-latency/vmlatency/internal/latency/pingparser_test.go:70]: If there is only one test-case, IMO we can drop the table (pingParserTestCase + for loop).

File: checkups/kubevirt-vm-latency/vmlatency/internal/latency/pingparser_test.go (hunk 1 above)
Comment (orelmisan) [inline checkups/kubevirt-vm-latency/vmlatency/internal/latency/pingparser_test.go:31]: `pingOutput` could be moved to `TestParsePingResults()` in order to decrease its scope.

File: checkups/kubevirt-vm-latency/vmlatency/internal/latency/pingparser_test.go [hunk 4]
```diff
@@ -42,24 +40,33 @@ PING 1.1.1.1 (1.1.1.1) 56(84) bytes of data.
...
-	if expectedResults.Time, err = time.ParseDuration("4004ms"); err != nil {
-		panic(err)
-	}
-	if expectedResults.Min, err = time.ParseDuration("1.732ms"); err != nil {
-		panic(err)
-	}
-	if expectedResults.Average, err = time.ParseDuration("2.074ms"); err != nil {
-		panic(err)
+
+var pingResults = latency.Results{
```
Comment (orelmisan) [inline checkups/kubevirt-vm-latency/vmlatency/internal/latency/pingparser_test.go:44]: `pingResults` could be moved to `TestParsePingResults()` in order to decrease its scope.

**Review by orelmisan — COMMENTED** [review]
Thank you for the PR @ormergi!
Please see the inline comments.

File: checkups/kubevirt-vm-latency/vmlatency/internal/latency/pingparser_test.go (hunk 4 above)
Comment (orelmisan) [inline checkups/kubevirt-vm-latency/vmlatency/internal/latency/pingparser_test.go:44]: `pingResults` could be moved to `TestParsePingResults()` in order to decrease its scope.

File: checkups/kubevirt-vm-latency/vmlatency/internal/latency/pingparser_test.go (hunk 3 above)
Comment (ormergi) [reply checkups/kubevirt-vm-latency/vmlatency/internal/latency/pingparser_test.go:70]: I added the table as preparation step for follow up PR that adds more tests to it.

File: checkups/kubevirt-vm-latency/vmlatency/internal/latency/pingparser_test.go (hunk 1 above)
Comment (ormergi) [reply checkups/kubevirt-vm-latency/vmlatency/internal/latency/pingparser_test.go]: It left out of the function to not bloat it when other cases are added.

File: checkups/kubevirt-vm-latency/vmlatency/internal/latency/pingparser_test.go (hunk 4 above)
Comment (ormergi) [reply checkups/kubevirt-vm-latency/vmlatency/internal/latency/pingparser_test.go]: Same here, I left it out of the function to not bloat it when other cases are added.

**Comment by ormergi** [comment]
[Changes](https://github.com/kiagnose/kiagnose/compare/7343f30952dce79dc454115077fa5903f4b5870b..ca845f28cfd8fa99ae83baf49f1a7c1e16f79864): Add more tests.

File: checkups/kubevirt-vm-latency/vmlatency/internal/latency/pingparser_test.go [hunk 5]
```diff
@@ -50,6 +74,34 @@ var successfulPingResults = latency.Results{
...
+`
+
+var pingResultsWithoutLatencyInfo = latency.Results{
+	Transmitted: 5,
+	Received:    5,
+	Time:        time.Millisecond * 4004,
+}
+
+const pingOutputWithoutPacketsInfo = `
+rtt min/avg/max/mdev = 1.732/2.074/2.382/0.214 ms
```
Comment (RamLavi) [inline checkups/kubevirt-vm-latency/vmlatency/internal/latency/pingparser_test.go:96]: is this a thing?

**Review by RamLavi — APPROVED** [review]
L2xndG0=

**Review by EdDev — CHANGES_REQUESTED** [review]
> vmlatency, ping-parser: Improve test coverage 

Please add more information about what scenarios you tests and what these represent.
I for example am not clear why you check duplication or why partial info is ok.

**Comment by EdDev** [comment]
@ormergi , is this change relevant?

**Comment by ormergi** [comment]
> @ormergi , is this change relevant?

Nope, I am closing it.

**Pull request closed without merging**